packages/node_modules/@webex/plugin-meetings/src/peer-connection-manager/index.js

```diff
@@ -310,50 +354,60 @@ pc.setRemoteSessionDetails = (peerConnection, typeStr, remoteSdp, meetingId) =>
       }
     });
   }
-  if (peerConnection.signalingState === SDP.HAVE_LOCAL_OFFER || (peerConnection.signalingState === SDP.STABLE && typeStr === SDP.OFFER)) {
-    return peerConnection.setRemoteDescription(
-      new window.RTCSessionDescription({
-        type: typeStr,
-        sdp
-      })
-    )
-      .then(() => {
-        if (!peerConnection.remoteDescription) {
-          Metrics.postEvent({
-            event: eventType.REMOTE_SDP_RECEIVED,
-            meetingId
-          });
-        }
-      })
-      .catch((error) => {
-        LoggerProxy.logger.error(`Peer-connection-manager:index#setRemoteDescription --> ${error} missing remotesdp`);
 
+  if (typeStr === SDP.OFFER && peerConnection.signalingState !== SDP.STABLE) {
+    return enqueueSdpOffer(peerConnection, SDP_OFFER_TYPE.REMOTE, {remoteSdp, meetingId});
+  }
 
-        const metricName = METRICS_OPERATIONAL_MEASURES.PEERCONNECTION_FAILURE;
-        const data = {
-          correlation_id: meetingId,
-          reason: error.message,
-          stack: error.stack
-        };
-        const metadata = {
-          type: error.name
-        };
+  if (typeStr === SDP.ANSWER && peerConnection.signalingState !== SDP.HAVE_LOCAL_OFFER) {
+    LoggerProxy.logger.error(`Peer-connection-manager:index#setRemoteSessionDetails --> received remote SDP in wrong state: ${peerConnection.signalingState}`);
 
-        Metrics.sendOperationalMetric(metricName, data, metadata);
+    return Promise.reject(new MediaError(`PeerConnection in wrong state: ${peerConnection.signalingState}`));
+  }
 
-        return Metrics.postEvent({
+  return peerConnection.setRemoteDescription(
+    new window.RTCSessionDescription({
+      type: typeStr,
+      sdp
+    })
+  )
+    .then(() => {
+      if (!peerConnection.remoteDescription) { // todo: why this condition is like this? shouldn't it be the opposite?
+        Metrics.postEvent({
           event: eventType.REMOTE_SDP_RECEIVED,
-          meetingId,
-          data: {
-            canProceed: false,
-            errors: [Metrics.generateErrorPayload(2001, true,
-              error.name.MEDIA_ENGINE, 'missing remoteSdp')]
-          }
+          meetingId
         });
-      });
-  }
+      }
+      if (typeStr !== SDP.OFFER) {
+        processNextQueuedSdpOffer(peerConnection);
+      }
+    })
+    .catch((error) => {
+      LoggerProxy.logger.error(`Peer-connection-manager:index#setRemoteDescription --> ${error} missing remotesdp`);
+
+
+      const metricName = METRICS_OPERATIONAL_MEASURES.PEERCONNECTION_FAILURE;
+      const data = {
+        correlation_id: meetingId,
+        reason: error.message,
+        stack: error.stack
+      };
+      const metadata = {
+        type: error.name
+      };
 
-  return Promise.reject(new MediaError('PeerConnection in wrong state'));
```

Comment:
Did you mean to stop propagating this error?

Reply:
yes, because now for the cases where we were returning this error we would just queue the request.